### ISSUE TITLE
Eager-loaded elements not provided as an array when 'one' => true

### ISSUE THREAD

**Issue by lukeyouell** [craftcms/element-api#70]
Any idea why this doesn't work with eager-loaded elements when `one` is set to `true`?

```php
'api/v1/code/<code>' => function($code) {
    return [
        'elementType' => Entry::class,
        'criteria' => [
            'section' => 'countries',
            'alpha2Code' => $code,
            'with' => ['capital'],
        ],
        'one' => true,
        'transformer' => function(Entry $entry) {
            $capital = ArrayHelper::firstValue($entry->capital);

            return [
                'capital' => $capital ? [
                    'name' => $capital->title,
                    'lat' => $capital->map->lat ? (float) $capital->map->lat : null,
                    'lng' => $capital->map->lng ? (float) $capital->map->lng : null,
                ] : null,
            ];
        },
    ];
}
```

`'one' => false` returns:
```json
"capital": {
  "name":"Canberra",
  "lat":-35.2809368,
  "lng":149.1300092
}
```

`'one' => true` returns:
```
Argument 1 passed to craft\\helpers\\ArrayHelper::firstValue() must be of the type array, object given
```

**Comment by bro-strummer** [comment]
@lukeyouell I can confirm this unexpected behavior as well!

The below endpoint – shortened for brevity – attempts to eager load assets in a matrix field block. When `'one' => false`, `print_r($block->image)` prints an array of Asset Objects as expected.

Unexpectedly, when `'one' => true`, `print_r($block->image)` prints an AssetQuery Object.

Knowing this, I've just been cruising along, leaving `'one' => false` and adding `'limit' => 1` to the `criteria` block of endpoints that utilize eager-loading and intend to return a single element. Eager-loading is intact, and the bloat of the meta block in the response isn't a super big deal (afaik).

```php
<?php

use craft\elements\Entry;

return [

    'endpoints' => [        

        /**
         * Individual Project Endpoint
         */
        'api/v1/projects/<slug:{slug}>.json' => function($slug) {            

            return [                
                'criteria' => [
                    'section' => 'projects',
                    'slug' => $slug,                    
                    'with' => [                            
                        ['projectPieces.image:image'],          // asset field                                      
                    ]
                ],
                'elementType' => Entry::class,
                // 'one' => true,                       // seemingly breaks eager-loading
                'transformer' => function(Entry $entry) {              

                    $pieces = [];

                    // loop through project pieces matrix
                    foreach ($entry->projectPieces as $block) {
                        switch ($block->type->handle) {
                                         
                            // image matrix block
                            case 'image':

                                // `print_r($block->image)` returns an array of Asset Objects when 'one' => false
                                // `print_r($block->image)` returns an AssetQuery Object when 'one' => true

                                $image = $block->image[0];
                                $optimizedImages = $image->workImageVariants;
                            
                                // image piece
                                $piece = [
                                    'id' => $image->id,
                                    'title' => $image->title,
                                    'type' => 'image',
                                ];                                                                                           

                                if ($block->caption) {
                                    $piece['caption'] = $block->caption;
                                }

                                array_push($pieces, $piece);
                                
                                break;
                        }
                    }
                    
                    return [
                        'id' => $entry->id,
                        'title' => $entry->title,
                        'url' => $entry->url,
                        'status' => $entry->status,
                        'body' => $entry->body,
                        'pieces' => $pieces,                      
                    ];
                }                
            ];
        },
    ]
];
```

**Comment by ajoliveau** [comment]
Hi !
I've ran into this problem too. Is there any news ?

**Comment by folkrav** [comment]
Can confirm the issue as well.

**Comment by brandonkelly** [comment]
This should have been fixed in Craft 3.0.10 (via 4d6892b048c598db8e430f30e8c369728b1ea2b7).
